packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx

```diff
@@ -0,0 +1,74 @@
+---
+description: 'Disallow using confusing built-in primitive class wrappers.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-wrapper-object-types** for documentation.
+
+The JavaScript language includes a set of primitives such as `boolean` and `number`s.
+JavaScript also contains a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
+TypeScript represents the primitive in lower-case types and instances of the wrapper classes with upper-case.
+
+Using the primitives like `0` instead of class constructors like `new Number(0)` is generally considered a JavaScript best practice.
+As a result, using the lower-case primitive type names like `number` instead of class names like `Number` is generally considered a TypeScript best practice.
+
+Similarly, TypeScript's `Function` type refers to objects created with the `Function` constructor, i.e. `new Function("eval'd code with implicit args")`.
+`Function` allows being called with any number of arguments and returns type `any`.
+It's generally better to specify function parameter and return types with function type syntax.
```

Comment:
```suggestion
The JavaScript language has a set of language types, but some of them correspond to two TypeScript types, which look similar: `boolean`/`Boolean`, `number`/`Number`, `string`/`String`, `bigint`/`BigInt`, `symbol`/`Symbol`, `object`/Object`.
The difference is that the lowercase variants are compiler intrinsics and specify the actual _runtime types_ (that is, the return value when you use the `typeof` operator), while the uppercase variants are _structural types_ defined in the library that can be satisfied by any user-defined object with the right properties, not just the real primitives.
JavaScript also has a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
These wrapper objects are assignable to the uppercase types, but not to the lowercase types.

Using the primitives like `0` instead of object wrappers like `new Number(0)` is generally considered a JavaScript best practice.
In general, your functions want to work with real numbers, instead of objects that "look like" numbers.
As a result, using the lower-case primitive type names like `number` instead of class names like `Number` is generally considered a TypeScript best practice.

TypeScript's `Function` type suffers from the same issue of structural typing.
In addition, `Function` allows being called with any number of arguments and returns type `any`, while in reality, it may be a class which is never callable, or it may be a plain object that happens to possess all properties of the `Function` class!
It's generally better to specify function parameters and return types with the function type syntax. If you want a generic "callable" type, use `(...args: never) => unknown`.
```